src/Libraries/Microsoft.Extensions.AI.Ollama/OllamaChatClient.cs

```diff
@@ -247,11 +253,36 @@ private static ChatMessage FromOllamaMessage(OllamaChatResponseMessage message)
         return new ChatMessage(new(message.Role), contents);
     }
 
+    private static FunctionCallContent ToFunctionCallContent(OllamaFunctionToolCall function)
+    {
+        var id = Guid.NewGuid().ToString().Substring(0, 8);
+        return new FunctionCallContent(id, function.Name, function.Arguments);
+    }
+
+    private static JsonElement? ToOllamaChatResponseFormat(ChatResponseFormat? format)
+    {
+        if (format is ChatResponseFormatJson jsonFormat)
+        {
+            if (!string.IsNullOrEmpty(jsonFormat.Schema))
+            {
+                return JsonDocument.Parse(jsonFormat.Schema!).RootElement;
```

Comment:
One option is to change `ChatResponseFormatJson.Schema` to be a `JsonElement?` instead of a `string?` so that we don't have to parse it here.

I'm not massively worried about the perf of parsing this, since it's only once-per-call and only in the Ollama native structured output case, this is more a question of what is the right abstraction. There isn't any question that the response format is JSON here (we already know it's a `ChatResponseFormatJson`); the only question is whether it's fair to force JSON schemas to be represented as a `JsonElement`. My sense is that yes, `JsonElement` is at least as reasonable a representation as `string` and is already what we do for the almost-equivalent function calling parameter schema.

@eiriktsarpalis / @stephentoub Do you have an opinion?

Reply:
One benefit of using `JsonElement` over `string` is that it works better when you're serializing it as a nested property. A string value will output the schema as a single escaped JSON string whereas `JsonElement` serializes as a nested value. Ultimately though, it depends on what pre-existing wire format you need to cater to.

Reply:
> JsonElement serializes as a nested value

In the Ollama case, we do want to serialize it as a nested value, which is what prompted this change.

In the OpenAI case, the API requires us to pass it as a serialized UTF8 value (in a `BinaryData`) so this change makes that slightly less convenient, but not enough to be problematic.

Anyway, this is now done.